src/anyio/_core/_sockets.py

```diff
@@ -345,15 +345,16 @@ def setup_raw_socket(
 
         return sock
 
-    # We passing type=0 on non-Windows platforms as a workaround for a uvloop bug
-    # where we don't get the correct scope ID for IPv6 link-local addresses when passing
-    # type=socket.SOCK_STREAM to getaddrinfo():
+    # We passing type=0 on non-Windows and non-illumos platforms as a
+    # workaround for a uvloop bug where we don't get the correct scope ID for
+    # IPv6 link-local addresses when passing type=socket.SOCK_STREAM to
+    # getaddrinfo():
     # https://github.com/MagicStack/uvloop/issues/539
     gai_res = await getaddrinfo(
         local_host,
         local_port,
         family=family,
-        type=socket.SOCK_STREAM if sys.platform == "win32" else 0,
+        type=socket.SOCK_STREAM if sys.platform in ("win32", "sunos5") else 0,
```

Comment:
Hmmm, this change is unrelated to the `port=0` issue.  It is just a fix to make this working properly (or better :-)) on illumos.  Maybe I should split out this to a new PR?